test/transport.js

```diff
@@ -0,0 +1,48 @@
+'use strict'
+
+const t = require('tap')
+const { once } = require('events')
+const { setTimeout } = require('timers/promises')
+const pino = require('pino')
+const { MongoClient } = require('mongodb')
+
+t.test('auth transport test', async (t) => {
+  const options = {
+    uri: 'mongodb://localhost:27017/',
+    database: 'logs',
+    collection: 'log-test',
+    mongoOptions: {
+      useNewUrlParser: true,
+      authMechanism: 'SCRAM-SHA-1',
+      auth: {
+        username: 'one',
+        password: 'two'
+      }
+    }
+  }
+
+  const client = new MongoClient(options.uri, options.mongoOptions)
+  await client.connect()
+  t.teardown(client.close.bind(client))
+  const db = client.db(options.database)
+  const collection = db.collection(options.collection)
+
+  const rowsBefore = await collection.countDocuments()
+
+  const transport = pino.transport({
+    target: '../pino-mongodb.js',
+    level: 'info',
+    options
+  })
+  const log = pino(transport)
+  await once(transport, 'ready')
+  log.info('this is a string log')
+  log.debug('ignored')
+  log.info('this is a long string log'.repeat(1000))
+  log.fatal(new Error('ops'), 'not ignored')
+  t.pass('logged on mongo')
+
+  await setTimeout(1000)
```

Comment:
Here, I did not found a way to wait `pino.flush()` (and it was triggering an error. I'm investigating)